pkg/libvirt/resolve_vm.go

```diff
@@ -99,3 +111,49 @@ func addToNotExistCache(pid uint64) {
 	}
 	cacheNotExist[pid] = false
 }
+
+func getMetadataVMID(vmid string) (string, error) {
+	metadataURI := config.GetLibvirtMetadataURI()
+	token := config.GetLibvirtMetadataToken()
+	domainName := ""
+
+	parts := strings.Split(vmid, "-")
+	if len(parts) < 3 {
+		return "", fmt.Errorf("the vmid provided doest not match machine-qemu-number-name : %v", vmid)
+	}
+
+	domainName = strings.Join(parts[3:], "-")
+
+	uri, _ := url.Parse(string(libvirt.QEMUSystem))
+	l, err := libvirt.ConnectToURI(uri)
+	if err != nil {
+		return "", fmt.Errorf("libvirt failed to connect: %v", err)
+	}
+
+	defer func() {
+		if err := l.Disconnect(); err != nil {
+			return
+		}
+	}()
```

Comment:
can't we `defer l.Disconnect()` instead?

Reply:
I did that but the linter says I don't check the return value of "l.Disconnect()"...

Reply:
we can silence the linter :) when we purposely ignore errors

Reply:
On second thought, how we return the error if Disconnect fails?

Reply:
I'm going to put the `defer l.Disconnect()` back into the code but the linter will failed on that part.

Reply:
@aharivel , I think we should return the error when running disconnect. A common pattern is to have something like 

```
func foo() (val, err error) {
   defer func() {
       err = l.Disconnect()
   }()
}
```

This way the caller gets to handle and report the error.

Reply:
Right, I do prefer this me too.